app/mailers/application_mailer.rb

```diff
@@ -22,6 +22,28 @@ class ApplicationMailer < ActionMailer::Base
   helper :application # gives access to all helpers defined within `application_helper`.
 
 
+  # if there is a problem communicating with the MailGun REST server, log the problem
+  def self.deliver_mail(mail)
+
+    begin
```

Comment:
Don't need `begin` since the `rescue` clause is at end of the method (less wordy this way).